### DISCUSSION TITLE
Contact problem

### DISCUSSION THREAD

**Discussion by QZheng82** [Anura3D/Anura3D_OpenSource#91] [Q&A] [answered]
I have a question, Figures 1 and 2 show the effect without and with contact turned on respectively, why do the material points fly around instead when the contact algorithm is turned on?Figure 3 and 4 show how I opened the contact.Is there a problem with the process? Or is it not necessary to turn on the contact algorithm.Thank you.
![SC51WJ8B6X LZJ 73{RAY1](https://user-images.githubusercontent.com/119284513/227776276-fe1a767e-20fa-49f1-82c7-4ab1159a01e0.png)
![~EA345$ BYJZMYB1$EI7Y0V](https://user-images.githubusercontent.com/119284513/227776286-47271354-080c-415c-8be2-295c5a9a6d1e.png)
![L3U QX{%5)5`NU@ ZOLX 01](https://user-images.githubusercontent.com/119284513/227776292-30ec84b6-b5ee-456b-9258-0d58b86beca5.png)
![934R6 KC}_MS{0FRJQ%VM7N](https://user-images.githubusercontent.com/119284513/227776301-2219b0e4-a56a-4b1b-9804-616cb7a692ce.png)

**Comment by FrancescaCeccato** [comment, marked as answer]
Maybe the structure is deforming too much? Keep in mind that mesh should always be aligned with the boundary of the structure. Consider using the moving mesh. 
If you want the pile to deform you cannot use the contact.

**Comment by QZheng82** [comment]
It is normal for piles to deform slightly under the action of the pile and soil. But if the deformation occurs without opening the contact, how is the contact algorithm between the pile and the soil to be achieved? Or is the material point method itself able to simulate this contact?

**Reply by FrancescaCeccato** [reply]
In standard MPM rough contact is automatic. If you want to add frictional properties to the contact, special numerical features, such as the contact algorithm are necessary. In Anura3D the contact surface is defined at the beginning of the simulation, therefore the structure cannot deform too much. This has several computational advantages (computational efficiency, accuracy...), but also some drawbacks such as the impossibility to follow large deformations of the structure.
I think that for your problem you can probably get good results without contact.

**Comment by EmreDumlu-UM** [comment]
the dt you used could be higher. Could you track the velocity particle? By using x/v=dt, you may be able to avoid abnormal points. the x here is the minimum element size.
